From 56dd84d0626f99f97ae94d868058367ee46d9a73 Mon Sep 17 00:00:00 2001
From: Ratchanan Srirattanamet <ratchanan@ubports.com>
Date: Fri, 22 Mar 2024 17:30:28 +0700
Subject: [PATCH] configure.ac: fix building against BusyBox sed

Welp, when you are not using `-E` (extended) mode, basic POSIX regex
means that `+` doesn't have to be escaped. Escaping it confuses some
version of sed, such as BusyBox sed in recent version of Alpine Linux
which fails with `sed: bad regex '\+': Repetition not preceded by valid
expression`.

POSIX regex, why do you have basic mode?

Fixes: https://gitlab.com/ubports/development/core/click/-/issues/26
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ff69711..f173da9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -154,7 +154,7 @@ AC_SUBST(COVERAGE_LDFLAGS)
 
 # Python can't accept some characters in our Debian version, so strip those out.
 AC_SUBST(PYTHON_PACKAGE_VERSION,
-	[$(echo "$PACKAGE_VERSION" | sed -e 's/[[^a-zA-Z0-9.+-]]//g' -e 's/\+//2g')])
+	[$(echo "$PACKAGE_VERSION" | sed -e 's/[[^a-zA-Z0-9.+-]]//g' -e 's/+//2g')])
 
 AC_CONFIG_FILES([
 	Makefile
-- 
2.44.0

